lib/ts/recipe/dashboard/api/userroles/roles/getAllRoles.ts

```diff
@@ -0,0 +1,50 @@
+import UserRoles from "../../../../userroles";
+import UserRolesRecipe from "../../../../userroles/recipe";
+
+import { APIFunction, APIInterface, APIOptions } from "../../../types";
+
+type Roles = Array<{ role: string; permissions: string[] }>;
+
+type Response =
+    | {
+          status: "OK";
+          roles: Roles;
+      }
+    | {
+          status: "FEATURE_NOT_ENABLED_ERROR";
+      };
+
+const getAllRoles: APIFunction = async (_: APIInterface, __: string, ___: APIOptions, ____: any): Promise<Response> => {
```

Comment:
Is this API only called when the user clicks on the roles management page on the dashboard? Cause this seems very expensive... If there are 100s of roles, and each have permissions, this API will take forever. Ideally, we don't want to fetch the permissions this way, and fetch them in a batch of like 10 roles at a time from the frontend.